src/commands/postDeploy/getLinuxDetectorError.ts

```diff
@@ -3,83 +3,97 @@
+    if (appInsightTable) {
+        insightTable = appInsightTable;
+        context.telemetry.properties.insight = 'app';
+        detectorTimestamp = getValuesByColumnName(context, appInsightTable, ColumnName.dataName).replace(bracketsAndSpace, '');
+    } else {
+        // if there are no app insights, defer to the Docker container
+        const dockerInsightTable: detectorTable | undefined = findTableByName(insightDataset, 'docker/insight');
+        if (!dockerInsightTable) {
             return undefined;
         }
 
-        const errorMessages: string | undefined = getValuesByColumnName(context, criticalErrorTable, ColumnName.message);
-        const failureCount: string | undefined = getValuesByColumnName(context, failureCountTable, ColumnName.failureCount);
+        insightTable = dockerInsightTable;
+        context.telemetry.properties.insight = 'docker';
+        detectorTimestamp = getValuesByColumnName(context, dockerInsightTable, ColumnName.dataName).replace(bracketsAndSpace, '');
+    }
 
-        context.telemetry.properties.errorMessages = JSON.stringify(errorMessages);
-        context.telemetry.properties.failureCount = failureCount;
+    detectorTimestamp = moment.utc(detectorTimestamp).format(timeFormat);
 
-        return `Critical insights found for "${siteName}": ${failureCount} as of ${new Date(timestamp).toLocaleString()}. Critical insight preview: "${errorMessages}". Click here to access "App Service Diagnostics" for recommendations.`;
+    if (!detectorTimestamp || !validateTimestamp(context, detectorTimestamp, deployEndTime)) {
+        return undefined;
+    }
+
+    if (getValuesByColumnName(context, insightTable, ColumnName.status) === 'Critical') {
+        const insightError: string = getValuesByColumnName(context, insightTable, ColumnName.dataValue);
+        context.telemetry.properties.errorMessages = JSON.stringify(insightError);
+        return localize('criticalError', '"{0}" reported a critical error: {1}', node.root.client.siteName, insightError);
     }
 
     return undefined;
 }
 
-export function validateTimestamp(context: IActionContext, detectorTime: Date, deployResultTime: Date): boolean {
-    const secondsBetweenTimes: number = Math.abs((detectorTime.getTime() - deployResultTime.getTime()) / 1000);
-
-    // we don't know how long it takes for the deployResult startTime and the detector timestamp
+export function validateTimestamp(context: IActionContext, detectorTime: string, deployResultTime: string): boolean {
+    const secondsBetweenTimes: number = (new Date(detectorTime).getTime() - new Date(deployResultTime).getTime()) / 1000;
+    // the detector can be as fast as ~20 seconds for app errors, but Docker container errors seem to timeout at
+    // about 5 minutes
     context.telemetry.properties.timeBetweenDeployAndDetector = secondsBetweenTimes.toString();
-    if (secondsBetweenTimes <= 60) {
+
+    // detector time must be more recent than deployResultTime
```

Comment:
So this is only an issue if they start another deploy while we're still checking the last deploy for detector errors. But this gets canceled if they start another deploy, right? I mean...at least if they're deploying from the same VS Code window

Reply:
Yeah, correct. We cancel it if we detect that the post-deployment task is going on.  But yeah, someone could screw it up if they deploy it from another window, but technically, they'd probably still want the most recent error anyway.